### PR TITLE
separate numbering for 2017 and 2023 workflows

### PR DESCRIPTION
**Pull request by kpedro88**
Due to an increasing number of conflicting upgrade PRs (#15662, #15613, #15805, with more coming), it was decided to separate the numbering for 2017 and 2023 workflows. The authors of the above PRs will have to rebase once this PR is merged.

The separate numbering is implemented by adding a new dimension to the dictionaries used to construct the upgrade workflows. This foresees possible expansions, e.g. using 30000 for 2019 workflows.

attn: @davidlange6

### CODE REVIEW

**Comment by kpedro88** [comment]
@cmsbuild please test

**Comment by fabozzi** [comment]
@piperov 
this is of interest for you

**Comment by fabozzi** [comment]
+1

**Comment by davidlange6** [comment]
+1

File: Configuration/PyReleaseValidation/scripts/runTheMatrix.py [hunk 1]
```diff
@@ -59,8 +59,8 @@ def runSelected(opt):
                      135.4, #Run 2 Zee ttbar
                      10021.0, #2017 tenmu
                      10024.0, #2017 ttbar
-                     10424.0, #2023D1 ttbar (Run2 calo)
-                     11224.0, #2023D3 ttbar (HGCal)
+                     20424.0, #2023D1 ttbar (Run2 calo)
+                     21224.0, #2023D3 ttbar (HGCal)
```
Comment (slava77) [inline Configuration/PyReleaseValidation/scripts/runTheMatrix.py]: The workflow list is incorrect or the numbering has to be redone
20424.0 is ```TTbar_13TeV_TuneCUETP8M1_2023D2```
and 21224.0 does not exist.
It should be 20024.0 and 20824.0 based on the numbering generated (I'm relying on CMSSW_8_1_X_2016-09-13-1100)

**Pull request merged**